ansible_collections/arista/cvp/plugins/module_utils/tools_cv.py

```diff
@@ -66,10 +66,14 @@ def cv_connect(module):
     ansible_connect_timeout = connection.get_option(
         "persistent_connect_timeout")
 
-    if not isinstance(user_authentication, str):
+    # The following is needed because of https://github.com/ansible/ansible/issues/75503
+    # Which was fixed in https://github.com/ansible/ansible/pull/78236
+    # This is a failsafe in case the ansible version is not high enough to have the fix
+    if isinstance(user_authentication, dict) and "__ansible_vault" in user_authentication:
         LOGGER.error('Cannot connect to CVP, password is encrypted')
-        raise NotImplementedError("Vault encrypted variables are not supported "
-                                  "as password yet. Use ansible vault file instead."
+        raise NotImplementedError("Vault encrypted variables are not supported with as password with your version of ansible. "
```

Comment:
nit: should that `Vault encrypted variables are not supported with as password with your version of ansible.` be
`Vault encrypted variables are not supported for passwords with your version of ansible.` ?

Reply:
fixed in cb34ebb

Thanks!